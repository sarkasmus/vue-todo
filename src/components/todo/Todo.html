<div class="todo">
  <div class="row">
    <div class="col-md-1 control">
      <label class="control control-checkbox">
        <input type="checkbox" :checked="finished" @change="finish" class="finish">
        <div class="control_indicator"></div>
      </label>
    </div>
    <div class="col-md-6">
      <h3 class="title" :title="description">{{ title }}</h3>
    </div>
    <div class="col-md-2">
      <span>{{ date.getMonth() | month }} {{ date.getDate() }}, {{ date.getFullYear() }}</span>
    </div>
    <div class="col-md-3 text-right">
      <button class="remove btn btn-danger btn-sm" @click="remove">Delete</button>
      <button class="remove btn btn-warning btn-sm" @click="edit">Edit</button>
    </div>
  </div>
</div>
