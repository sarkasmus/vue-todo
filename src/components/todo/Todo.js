export default {
  name: 'Todo',
  props: {
    title: {
      type: String
    },
    description: {
      type: String
    },
    finished: {
      type: Boolean
    },
    dueDate: {
      type: String
    }
  },
  data () {
    return {
      date: new Date(this.dueDate)
    }
  },
  methods: {
    finish () {
      this.$emit('finished')
    },

    remove () {
      this.$emit('removed')
    }
  },
  filters: {
    day (day: number) {
      let days = [
        'Sunday',
        'Monday',
        'Tuesday',
        'Wednesday',
        'Thursday',
        'Friday',
        'Saturday'
      ]

      return days[day]
    },

    month (month: number) {
      let months = [
        'January',
        'February',
        'March',
        'April',
        'May',
        'June',
        'July',
        'August',
        'September',
        'October',
        'November',
        'December'
      ]

      return months[month]
    }
  }
}
