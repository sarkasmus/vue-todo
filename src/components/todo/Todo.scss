@import "_checkbox.scss";

.title {
  font-size: 22px;
  font-weight: bold;
  color: #666666;
}
