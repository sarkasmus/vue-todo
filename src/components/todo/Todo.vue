<template src="./Todo.html"></template>
<style src="./Todo.scss"></style>
<script src="./Todo.js"></script>
