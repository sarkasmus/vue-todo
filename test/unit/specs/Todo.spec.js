import { mount } from '@vue/test-utils'
import Todo from '../../../src/components/todo/Todo.vue'

/**
 * Testing component representing single task (todo).
 */
describe('Todo', () => {
  let todoComponent

  /**
   * Inits every testcase.
   */
  beforeEach(() => {
    todoComponent = mount(Todo, {
      propsData: {
        title: 'Task 1',
        description: 'My awesome task',
        finished: false,
        dueDate: '2018-06-02'
      }
    })
  })

  it('should be able to check and uncheck finished and emit an event', () => {
    todoComponent.find('.finish').trigger('click')

    expect(todoComponent.emitted().finished).toBeTruthy()
  })

  it('should be able to delete itself and emit an event', () => {
    todoComponent.find('.remove').trigger('click')

    expect(todoComponent.emitted().removed).toBeTruthy()
  })

  /**
   * Matching snapshot - HTML structure of the component.
   */
  it('should match snapshot', () => {
    expect(todoComponent.element).toMatchSnapshot()
  })
})
