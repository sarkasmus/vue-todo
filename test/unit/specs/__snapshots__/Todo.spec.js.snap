// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match snapshot 1`] = `
<div
  class="todo"
>
  <div
    class="row"
  >
    <div
      class="col-md-1 control"
    >
      <label
        class="control control-checkbox"
      >
        <input
          class="finish"
          type="checkbox"
        />
         
        <div
          class="control_indicator"
        />
      </label>
    </div>
     
    <div
      class="col-md-6"
    >
      <h3
        class="title"
        title="My awesome task"
      >
        Task 1
      </h3>
    </div>
     
    <div
      class="col-md-2"
    >
      <span>
        June 2, 2018
      </span>
    </div>
     
    <div
      class="col-md-3 text-right"
    >
      <button
        class="remove btn btn-danger btn-sm"
      >
        Delete
      </button>
       
      <button
        class="remove btn btn-warning btn-sm"
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
